src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -280,6 +314,295 @@ size_t runtime::gpu::CUDAEmitter::build_1d_max_pool(const GPURuntimeContext* ctx
     return primitive_index;
 }
 
+pooling_op_shape avgpool_shape(
+    const Shape& in, const Shape& out, const Shape& window, const Shape& strides, const Shape& pad)
+{
+    pooling_op_shape shape;
+    shape.N = static_cast<int>(in[0]);
+    shape.C = static_cast<int>(in[1]);
+    shape.K = shape.C; // pooling feature maps is
+    shape.J = shape.C; // not currently supported
+    if (in.size() == 3)
+    {
+        shape.D = 1;
```

Comment:
You are correct, this would need some additional engineering to move to N dimensions. However, from what I can tell, the frameworks are typically using only 2 or 3 dimensions. For example, Tensorflow only supports 2/3d for average pooling 

* https://www.tensorflow.org/api_docs/python/tf/nn/avg_pool
* https://www.tensorflow.org/api_docs/python/tf/nn/avg_pool3d

Reply:
I see. Let's not worry it for now.